### PR TITLE
Update README to prepare for GR4.0 incorporation

### PR DESCRIPTION
**Pull request by cafeclimber**
First swing at updating the README to prepare for incorporating the repo into the GR organization

### CODE REVIEW

**Comment by RalphSteinhagen** [comment]
@cafeclimber thanks for this PR! I think this is an excellent start. :+1: 

Some thoughts I'd like to share for further enhancements:

1. **Logo Update Proposal**
   - How about updating the logo to include `GNU Radio'^{4.0 - beta0}`? This could help users quickly identify the version and distinguish it from previous versions. This style was effectively used in a previous FAIR-specific announcement.
  <details>
  <summary>Image example: click to expand!</summary>
  
   gnuradio_logo
   gnuradio_logo

  </details>

2. **Introduction of GR4 Features**
   - I propose we add a brief section outlining the key features, goals, and improvements in GR4 right at the top, something like this:
     ```
     ## What's New in GNU Radio 4.0?
     - **Seamless Transition and Preserving Established GR3 Paradigms**: same user-defined block and flow-graph topologies accessible graphically, via Python, and using C++.
     - **Enhanced Data Types**: support for fundamental data types (i.e. ints, floats, complex numbers) as well as more complex, structured, and user-defined custom types.
     - **Simplified Block Development**: utilising modern C++ and computing standards for high-performance.
     - **High-Performance Signal-Processing**: significant boosts from lock-free buffers, compile-time optimisations, and built-in SIMD & SYCL support.
     - **Flexible Scheduling**: shift from a 'micro-service' style in GR3 to a work-based scheduling scheme in GR4, facilitating user-defined custom schedulers that enable trade-off between throughput, parallelism, and latency (N.B. maybe a bit too technical)
     - **Recursive Directed Graphs**: support for basic feedback loops.
     - **Broadened Hardware Support**: ready for CPUs, MCUs, GPUs, and potentially FPGAs. (N.B. the 'FPGA' is a bit more aspirational at this time, since we do not have an PoC yet, but SYCL enables this for both [Intel](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/C++SYCL_FPGA/Tutorials/GettingStarted/fpga_compile) and [AMD/Xilinx](https://github.com/triSYCL/sycl/blob/sycl/unified/master/sycl/doc/GettingStartedXilinxFPGA.md) FPGAs)
     - **Bridging R&D and Industrial Use**: aims at bridging the gap and know-how transfer between R&D prototyping by academics, hobbyists and safe operational use by research organisations and industry.
     - [More on architecture and design](https://github.com/fair-acc/graph-prototype/tree/main/core)
     ```

3. **Technical and Demo Additions Post-Merge**
   - More technical details, code examples, and demos will be added after this initial merge to keep our README clean yet informative.

4. **Open Questions**
   - Copyright and Acknowledgments Revision
     * Should we update the copyright to include already the GNU Radio organisation explicitly now? 
     * How should we register and acknowledge the extensive contributions from companies, universities, research organisations, and individual users? Perhaps an automated system to update the contributors list directly from GitHub could streamline this process.
     * 'Acknowledgments' could include details of various grants and organisations that have or are supporting this project.

You feedback would be appreciated. Particularly, are there additional features or improvements we should highlight?

**Comment by RalphSteinhagen** [comment]
... we should also comment on that issues regarding GR4 should be posted in this GH repo and the ones regarding in the GR3 repo... just to keep things separate (for now). Later, as soon as many have moved towards GR4 we could shift this to the main org/repo.

**Review by RalphSteinhagen — APPROVED** [review]

**Comment by mormj** [comment]
I need to get a bit more clarification on our ability as an org for copyright assignment, but for GR3 (and assuming for GR4 as well), we follow the DCO https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed and copyright for contributions remain (or can remain) with the author.  So keeping a copyright list here in the README will get complicated over time. Same for the contributors - if we can note the originating contributors in the acknowledgements and attribution to FAIR that might be simpler.

I'm thinking:
- License (SPDX LGPL etc. unless otherwise noted)
- Copyrights (brief explanation of the DCO)
- Acknowledgements (include key GSI/FAIR contributors)

**Comment by mormj** [comment]
```
## License and Copyright
Any code contributions going into GNU Radio will become part of an LGPL-licensed, open source repository, and copyright assignment remains with the author or organization supporting the work.

Unless otherwise noted: [SPDX-License-Identifier: LGPL-3.0-or-later](https://spdx.org/licenses/LGPL-3.0-or-later.html)
Copyright (C) 2018-2024 FAIR -- Facility for Antiproton & Ion Research, Darmstadt, Germany<br/>

## Acknowledgements
The GNU Radio project extends its profound gratitude to FAIR team for their remarkable contributions that have been instrumental in the development of GNU Radio 4.0. We are profoundly thankful for their invaluable efforts in advancing the frontiers of open-source SDR.  Specifically we would like to acknowledge the following contributors that built the fundamentally redesigned core architecture that has become GR 4.0:

 * Ivan Čukić <ivan.cukic@kdab.com>
 * Matthias Kretz <M.Kretz@GSI.de>
 * Alexander Krimm, <A.Krimm@GSI.de> 
 * Ralph J. Steinhagen, <R.Steinhagen@GSI.de>
...
```

**Comment by RalphSteinhagen** [comment]
@mormj, we greatly appreciate your recognition of FAIR's role in the development of GNU Radio 4.0. We are standing on the shoulders of giants. This project has been a collaborative effort, building on the foundational work laid by the original authors and the continuous innovations contributed by the GNU Radio [community](https://github.com/gnuradio/gnuradio/graphs/contributors) beyond GSI/FAIR. Many have paved the way, and software development is as much about the actual technology—where we implement specific high-performance and C++ modernization practices in the hope of fostering long-term maintenance and growth—as it is about the people and community that jointly build and groom that technology. The experiences with GR3 and many iterations during the GR4 architecture meetings, online discussions, and conferences have been and are key to this.

Thus, in that light, we might tone down the following:

> The GNU Radio project extends its profound gratitude to the FAIR team for their remarkable contributions that have been instrumental in the development of GNU Radio 4.0. We are profoundly thankful for their invaluable efforts in advancing the frontiers of open-source SDR. Specifically, we would like to acknowledge the following contributors who built the fundamentally redesigned core architecture that has become GR 4.0:

to perhaps:

```
The GNU Radio project appreciates the contributions from FAIR in the co-development of GNU Radio 4.0. Their dedicated efforts have played a key role in enhancing the capabilities of our open-source SDR technology. 
We would like to recognize the following contributors for their roles in redesigning the core that has evolved into GR 4.0:
```

@cafeclimber, if you could also please add the [DCO](https://github.com/gnuradio/gnuradio/blob/main/DCO.txt) and (in alphabetical order):

```md
  * Frank Osterfeld, Frank.Osterfeld@kdab.com
  * Semen Lebedev, S.Lebedev@GSI.de
``` 

to the list of those who have been instrumental in this as well and joined after we posted the initial version of the old README.md.

@mormj, we will add the corresponding GitHub action to ensure that each commit is signed.

@cafeclimber, @mormj if we are in agreement, this PR could be moved from 'DRAFT' to 'for review'. Should we look into more/wider iterations or just merge and adapt afterwards, if needed?


@cafeclimber, thanks a lot again for drafting and making these changes (I owe you both a :beer: :wine_glass: during the next GRCon). Much appreciated! :+1:

**Comment by wirew0rm** [comment]
Nice work :+1: 

One small thing, there was some discussion on how the list of contributors should be ordered. I think maybe this would be better represented as a table, e.g. [here](https://github.com/all-contributors/all-contributors?tab=readme-ov-file#contributors-) (Just the presentation as a table, I also like the emoji legend things, but i think that's more effort than value at the moment).

**Comment by RalphSteinhagen** [comment]
@cafeclimber some change request based on some internal discussions:

  * This one is from @ivan-cukic - if you could change:
    > **Seamless Transition by Maintaining GR3 Paradigms**: *[..]*

    to    
    >**Smooth Transition with Updated GR3 Paradigms**: User-defined blocks and flow-graphs continue to be accessible graphically, through Python, and using C++. Notable simplifications and streamlining have been made to the underlying low-level architecture and design, as described [here](https://github.com/fair-acc/graph-prototype/tree/main/core).

    and remove the last bullet point, 'More on architecture and design'

  * Could you make the following slight change from
    > This is the repository containing the prototype of GNU Radio 4.0. It is currently in a beta state and not ready for production use. The main organization and repository can
    
    to:
    > This is the GNU Radio 4.0 (GR4) prototype and is currently in a beta state. For production use, please use the GNU Radio 3.X (GR3) version found [here](https://github.com/gnuradio/gnuradio). [..]
    
    Just to keep it a bit shorter and to avoid having to redirect all the links in the (hopefully very near) future.
    
  * simplify the 'License and Copyright' legalese by referring directly to the '[CONTRIBUTING.md](CONTRIBUTING.md)', i.e.
    > Unless otherwise noted: [SPDX-License-Identifier: LGPL-3.0-or-later](https://spdx.org/licenses/LGPL-3.0-or-later.html)
    > All code contributions to GNU Radio will be integrated into a library under the LGPL, ensuring it remains free/libre (FLOSS) for both personal and commercial use, without further constraints on either.
    > For details on how to contribute, please consult: [CONTRIBUTING.md](https://app.element.io/CONTRIBUTING.md)
    > 
    > Copyright (C) 2001-September 2020 GNU Radio Project -- managed by Free Software Foundation, Inc.
    > Copyright (C) September 2020-2024 GNU Radio Project -- managed by SETI Institute
    > Copyright (C) 2018-2024 FAIR -- Facility for Antiproton & Ion Research, Darmstadt, Germany
    
    @mormj if you could confirm so that @cafeclimber could incorporate this with confidence.

I very much like the idea above by @wirew0rm we should IMO transit to that soon. The order of the contributors is meant to be alphabetically but by no means complete nor other things applied. @mormj had also an idea on how to separately track individuals and organisations that contributed. Since the latter list isn't complete, I'd propose to keep things simple and add the bits regarding the organisations backing/supporting/using GR later.

@cafeclimber thanks again for the revision. I initially thought this to be quick-and-easy but I think to let this ripe may make it better like a good red wine. If you could also move from 'DRAFT' to 'in review' so that we -- provided there are no formal objections -- could merge this by the end of this or latest by beginning of next week.

**Comment by mormj** [comment]
Yes, this all looks good!

**Comment by cafeclimber** [comment]
Mmk, apologies for the delay. Was traveling the last few days. Switching out of draft with the last commit 😄

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,6 +15,14 @@ examples or if you cannot find anything that fits your use-case use GitHub's dis
 ## Contributing code and content
 We welcome all forms of contributions from the community. Please read the following guidelines to maximise the chances of your PR being merged.
 
+### DCO Signing
+
+Any code contributions going into GNU Radio will become part of a GPL-licensed, open source repository. It is therefore imperative that code submissions belong to the authors, and that submitters have the authority to merge that code into the public GNU Radio codebase.
```
Comment (wirew0rm) [inline CONTRIBUTING.md:20]:
```suggestion
Any code contributions going into GNU Radio will become part of a (L)GPL-licensed, open source repository. It is therefore imperative that code submissions belong to the authors, and that submitters have the authority to merge that code into the public GNU Radio codebase.
```

File: README.md [hunk 2]
```diff
@@ -1,30 +1,65 @@
...
+energy particle physics, astrophysics, radio astronomy and more!
+
+## Helpful Links
+
+* [GNU Radio Website](https://gnuradio.org)
+* [GNU Radio Wiki](https://wiki.gnuradio.org/)
+* [Github issue tracker for bug reports and feature requests](https://github.com/fair-acc/graph-prototype/issues)
+* [View the GNU Radio Mailing List Archive](https://lists.gnu.org/archive/html/discuss-gnuradio/)
+* [Subscribe to the GNU Radio Mailing List](https://lists.gnu.org/mailman/listinfo/discuss-gnuradio)
+* [GNU Radio Chatroom on Matrix](https://chat.gnuradio.org/)
```
Comment (wirew0rm) [inline README.md:30]: The link between gnuradio4 and the architecture channel seems not really obvious... I guess people in gnuradio-general do a good job in redirecting the right way but we could just directly link to the channel.
```suggestion
* [GNU Radio Chatroom on Matrix](https://chat.gnuradio.org/)
  * Specifically for discussions related to GNURadio 4.0 join the [#architecture channel](https://matrix.to/#/#architecture:gnuradio.org)
```

**Review by wirew0rm — APPROVED** [review]
Looks great, thnaks a lot! Reads all very good, I've added 2 small comments, sorry for bringing stuff in last minute, not mad if you decide to just go ahead and merge.

As a sidenote I'd propose to skip the CI for docs only changes in the future: https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs. (OTOH that option will probably go away once we use an action to assemble the contributors list...)

**Pull request merged**